src/data/List.js

```diff
@@ -1159,23 +1159,31 @@ listProto.indicesOfNearest = function (dim, value, maxDistance) {
         maxDistance = Infinity;
     }
 
-    var minDist = Number.MAX_VALUE;
+    var minDist = Infinity;
     var minDiff = -1;
+    var nearestIndicesLen = 0;
     for (var i = 0, len = this.count(); i < len; i++) {
-        var diff = value - this.get(dim, i /*, stack */);
+        var diff = value - this.get(dim, i);
         var dist = Math.abs(diff);
-        if (diff <= maxDistance && dist <= minDist) {
-            // For the case of two data are same on xAxis, which has sequence data.
-            // Show the nearest index
-            // https://github.com/ecomfe/echarts/issues/2869
-            if (dist < minDist || (diff >= 0 && minDiff < 0)) {
+        if (dist <= maxDistance) {
+            // When the `value` is at the middle of `this.get(dim, i)` and `this.get(dim, i+1)`,
+            // we'd better not push both of them to `nearestIndices`, otherwise it is easy to
+            // get more than one item in `nearestIndices` (more specifically, in `tooltip`).
+            // So we chose the one that `diff >= 0` in this csae.
+            if (dist < minDist
+                || (dist === minDist && diff >= 0 && minDiff < 0)
+            ) {
                 minDist = dist;
                 minDiff = diff;
-                nearestIndices.length = 0;
+                nearestIndicesLen = 0;
+            }
+            if (diff === minDiff) {
```

Comment:
I'm confused about this condition. `diff` should be always same with  `minDiff` if https://github.com/apache/incubator-echarts/blob/2eac3913216f4f9a462897a0da839a878d438dea/src/data/List.js#L1177 is executed.

Reply:
The previous condition
```js
 if (dist < minDist
                || (dist === minDist && diff >= 0 && minDiff < 0) { ... }
```
is not always entered, as the previous comment described.

Reply:
Can we get the following rule from these codes:

If the condition 
```js
dist < minDist || (dist === minDist && diff >= 0 && minDiff < 0)
```
is failed.

Then

1. `dist < minDist` => `Math.abs(diff) !== Math.abs(minDiff)` => `diff !== minDiff`
2. Or `diff >= 0 && minDiff < 0 ` => `diff !== minDiff`

Which means the `diff === minDiff` condition will always be failed.

Reply:
`minDiff` and `minDist` will be modified if `dist < minDist || (dist === minDist && diff >= 0 && minDiff < 0)` is true. 

Take this test case for example:

```javascript
function core({diff, minDiff}) {
    console.log('---- minDiff:', minDiff, 'diff:', diff);
    var dist = Math.abs(diff);
    var minDist = Math.abs(minDiff);

    if (dist < minDist
        || (dist === minDist && diff >= 0 && minDiff < 0)
    ) {
        minDist = dist;
        minDiff = diff;
        nearestIndicesLen = 0;
        console.log('Clear previous');
    }
    if (diff === minDiff) {
        console.log('Add index');
    }
    else {
        console.log('Not add index');
    }
}


core({minDiff: 5, diff: -5})
core({minDiff: 5, diff: 5})
core({minDiff: -5, diff: 5})
core({minDiff: -5, diff: -5})
core({minDiff: 5, diff: 4})
core({minDiff: -5, diff: -4})
core({minDiff: 5, diff: 6})
core({minDiff: -5, diff: -6})
```

The result is:
```
---- minDiff: 5 diff: -5
 Not add index
---- minDiff: 5 diff: 5
 Add index
---- minDiff: -5 diff: 5
 Clear previous
 Add index
---- minDiff: -5 diff: -5
 Add index
---- minDiff: 5 diff: 4
 Clear previous
 Add index
---- minDiff: -5 diff: -4
 Clear previous
 Add index
---- minDiff: 5 diff: 6
 Not add index
---- minDiff: -5 diff: -6
 Not add index
```

Reply:
The key difference between your `core` function and test case is 
```js
```